### PR TITLE
No need for dstrcmp in object.d

### PR DESCRIPTION
**Pull request by andralex**
Just dogfood `__cmp`.

### CODE REVIEW

File: src/object.d [hunk 1]
```diff
@@ -328,14 +328,11 @@ class TypeInfo
 
     override int opCmp(Object o)
     {
-        import core.internal.string : dstrcmp;
-
-        if (this is o)
-            return 0;
+        assert(this !is o); // __cmp takes care of that
```
Comment (n8sh) [inline src/object.d:331]: I do not understand the reasoning here.

Comment (andralex) [reply src/object.d:331]: See [this line](https://github.com/dlang/druntime/blob/master/src/object.d#L4029). The call to `opCmp` comes from the lowering of `==`, which is `__cmp`.

Comment (n8sh) [reply src/object.d:331]: The lowering of `==` is `opEquals`. In any case isn't it valid to call `Object.opCmp` directly?

Comment (andralex) [reply src/object.d:331]: Sorry, I meant `a < b`. But maybe I am wrong?

Comment (rainers) [reply src/object.d:331]: `__cmp` takes care of the comparison, but you are disallowing `a < a` here before it is called.

Comment (andralex) [reply src/object.d:331]: @rainers OK OK I see what happens here. I assumed that https://github.com/dlang/dmd/pull/9629 and https://github.com/dlang/druntime/pull/2562 had been pulled.

**Pull request merged**